### DISCUSSION TITLE
Using resample_apply for indicators which require more than one column

### DISCUSSION THREAD

**Discussion by ahfattahi** [kernc/backtesting.py#440] [Q&A] [answered]
Hello,

Assume we have the hourly dataset of EURUSD.csv in the test section. I want to calculate the Stochastic Oscillator indicator in 4-hour steps. This indicator requires close, high, and low prices for each time step. I know I need to use [`reample_apply`](https://kernc.github.io/backtesting.py/doc/backtesting/lib.html#backtesting.lib.resample_apply), but I face errors. 
How should I use the resample_apply function if the indicator requires more than one column of data?

**Comment by kernc** [comment, marked as answer]
If you need to resample more than one column, you can pass in the whole dataframe instead of individual series:
```py
        def stoch_wrapper(df):
            import talib as ta
            return ta.STOCH(df.High, df.Low, df.Close)

        resample_apply('4H', stoch_wrapper, self.data.df)

**Comment by ahfattahi** [comment]
Thank you so much, your solution worked perfectly.
